### ISSUE TITLE
Unaligned 64-bit access causes crash on Hexagon platforms

### ISSUE THREAD

**Issue by petewarden** [tensorflow/tensorflow#39661]
We've discovered the cause of a TF Lite Micro crash on some Qualcomm Hexagon platforms, and need to figure out a good solution. Here are the details:

Quote from tensorflow/lite/micro/micro_allocator.cc:327 :

```

      // Note that the zero_point field in the FlatBuffers schema is a 64-bit

      // integer, but the zero_point field in the TfLiteQuantizationParams struct

      // is a 32-bit integer.

      result->params.zero_point =

          static_cast<int32_t>(src_quantization->zero_point()->Get(0));

```

As stated in the comment, the zero_point is a 64bit value inside flatbuffer’s schema.

But when hexagon DSP tries to read a 64bit value, the address needs to be 64bit aligned. This is a hardware instruction level limitation of our DSP.

So, when the execution fell into IndirectHelper::Read(), hexagon-sim crashed.

Quote from tensorflow/lite/micro/tools/make/downloads/flatbuffers/include/flatbuffers/flatbuffers.h:104 :

```

>>  static return_type Read(const uint8_t *p, uoffset_t i) {

      return EndianScalar((reinterpret_cast<const T *>(p))[i]);

    }

```

This pointer “p” is not guaranteed to be 8 byte aligned. So when I am not lucky, the linker will give it a bad location. Hence, random crash.

**Comment by njeffrie** [comment]
After evaluating options, I've opted to add alignas(8) to our serialized flatbuffers.  The entire flatbuffer is internally aligned properly (for example, 64-bit datatypes are aligned to 8 byte boundaries).

I have submitted a change to update our in-tree example and benchmark models to align the flatbuffer array to 8 bytes.  I am working to update alignment in models which are downloaded at compile time.
